File: c++/Jewels and Stones.cpp
#include <string>
#include <unordered_set>
#include <iostream>

class Solution {
public:
    int numJewelsInStones(std::string jewels, std::string stones) {
        std::unordered_set<char> jewelSet(jewels.begin(), jewels.end());
        int jewelCount = 0;
        for (char stone : stones) {
            if (jewelSet.count(stone) > 0) {
                jewelCount++;
            }
        }
        return jewelCount;
    }
};
//  test case
int main() {
    Solution sol;
    std::string jewels = "aA";
    std::string stones = "aAAbbbb";
    int count = sol.numJewelsInStones(jewels, stones);
    std::cout << "result of test case: " << count << std::endl;
    return 0;
}

File: python/Jewels and Stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        jewel_set = set(jewels)
        jewel_count = 0
        for stone in stones:
            if stone in jewel_set:
                jewel_count += 1
        return jewel_count